### DISCUSSION TITLE
RTK Query Cache Hit isRejected Detection Best Practice

### DISCUSSION THREAD

**Discussion by DavidPx** [reduxjs/redux-toolkit#1448] [Q&A]
Hi, my app uses a couple of RTK Query services and I'm using the following middleware for error handling duties.

I had to put in special logic to handle the "rejections" that come from cache hits - is there a better way to handle this?  Seeing the word rejected near a cache hit is odd.  Also I'm wondering if checking for `ConditionError` is the way to go.. I saw that in the Redux tools 🤷 

Thank you!

```
import {
	MiddlewareAPI,
	isRejectedWithValue,
	Middleware,
	isRejected,
} from '@reduxjs/toolkit';
import { pushToast } from '@ourstuff/toast';

// Show a toast upon error
export const rtkQueryErrorLogger: Middleware = (api: MiddlewareAPI) => (
	next,
) => (action) => {
	if (isRejectedWithValue(action)) {
		let msg = '';
		if (action.error?.message) {
			msg = action.error.message;
			if (action.payload?.status === 401) {
				msg += ': Invalid Token - Please refresh your browser';
			}
		} else {
			msg = JSON.stringify(action.error);
		}
		pushToast(msg);
	}
	// RTK Query cache hits
	else if (isRejected(action) && action.error?.name === 'ConditionError') {
		return next(action);
	} else if (isRejected(action)) {
		pushToast('Unknown error');
	}

	return next(action);
};
```

**Comment by phryneas** [comment]
Look for `action.meta.condition`. 

Generally, this is just using `createAsyncThunk` and `cAT` only knows "fulfilled" and "rejected", where "rejected" can either be an error thrown, a condition not applying (cache hit), a request being cancelled or a manual rejection with `rejectWithValue`.
